pkg/controllers/node/termination/terminator/eviction.go

```diff
@@ -205,6 +213,21 @@ func (q *Queue) Evict(ctx context.Context, key QueueKey) bool {
 		return false
 	}
 	NodesEvictionRequestsTotal.With(map[string]string{CodeLabel: "200"}).Inc()
-	q.recorder.Publish(terminatorevents.EvictPod(&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: key.Name, Namespace: key.Namespace}}))
+	q.recorder.Publish(terminatorevents.EvictPod(&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: key.Name, Namespace: key.Namespace}}, evictionMessage))
 	return true
 }
+
+func evictionReason(ctx context.Context, key QueueKey, kubeClient client.Client) (string, error) {
+	nodeClaim, err := node.NodeClaimForNode(ctx, kubeClient, &corev1.Node{Spec: corev1.NodeSpec{ProviderID: key.providerID}})
+	if err != nil {
+		return "", err
+	}
+	terminationCondition := nodeClaim.StatusConditions().Get(v1.ConditionTypeDisruptionReason)
+	if terminationCondition.IsTrue() {
+		return terminationCondition.Message, nil
+	}
+	if !nodeClaim.DeletionTimestamp.IsZero() {
```

Comment:
I think we don't need this check, we can just return "Forceful Deletion", nil if the condition doesn't exist.